refactor(parser): provide better error messages for `const` modifiers on class elements

DonIsaac opened the pull request:
Quality-of-life improvement for invalid `const` modifiers on class elements. Instead of panicking, the parser will eat `const` and report it as an error.

```ts
class C {
  static const H = 1;
}
```

DonIsaac commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6353?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6353** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6353?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#6352** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6352?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @DonIsaac and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

Boshen commented on tasks/coverage/snapshots/parser_typescript.snap:
```diff
@@ -4730,23 +4730,32 @@ Expect to Parse: tasks/coverage/typescript/tests/cases/conformance/salsa/private
+   ╭─[typescript/tests/cases/compiler/ClassDeclaration26.ts:2:18]
+ 1 │ class C {
+ 2 │     public const var export foo = 10;
+   ·                  ───
```
This is pointing at the wrong span.

DonIsaac replied:
I agree, I did not add checks for duplicate declaration keywords. I'm thinking this should go in another PR. Thoughts?

Boshen replied:
Sure.

DonIsaac committed:
refactor(parser): provide better error messages for `const` modifiers on class elements (#6353)

Quality-of-life improvement for invalid `const` modifiers on class elements. Instead of panicking, the parser will eat `const` and report it as an error.

```ts
class C {
  static const H = 1;
}
```
